src/core/qgsgeometryoptions.h

```diff
@@ -83,6 +85,38 @@ class CORE_EXPORT QgsGeometryFixes
      */
     void apply( QgsGeometry &geometry ) const;
 
+    /**
+     * A list of activated geometry checks.
+     *
+     * \since QGIS 3.4
+     */
+    QStringList geometryChecks() const;
+
+    /**
+     * A list of activated geometry checks.
+     *
+     * \since QGIS 3.4
+     */
+    void setGeometryChecks( const QStringList &geometryChecks );
```

Comment:
So how are the geometry checks going to work? Can we have an enum with the available checks? It is difficult for users of the API to figure out what checks can be used and what are the valid configuration options...